.github/workflows/lint.yaml

```diff
@@ -0,0 +1,119 @@
+name: lint
+on: [push, pull_request]
+env:
+  GH_ACTION: "TRUE"
+jobs:
+  check-licences:
+    name: Check Licences
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10' 
+      - name: Check licenses
+        shell: bash
+        run: ./scripts/enforce_licence.py
+  generic-nolints:
+    name: Count generic NOLINTs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Count the number of generic NOLINTs
+        shell: bash
+        run: ./scripts/count_generic_nolints.sh
+  build:
+    name: Build for Lint Tasks
+    runs-on: ubuntu-latest
+    container: debian:stable
+    steps:
+      - uses: actions/checkout@v3
+      - name: Update package database
+        run: apt-get update -y
+      - name: Install build tools
+        shell: bash
+        run: |
+          apt-get -y install pkg-config libtool autoconf \
+            automake g++ bison flex make bash-completion dh-autoreconf \
+            debhelper devscripts wget python3-pip
+      - name: Install Python lint tools
+        run: python3 -m pip install --no-input cpplint flake8
+      - name: Install build dependencies
+        shell: bash
+        run: |
+          apt-get -y install libcppunit-dev uuid-dev libncurses5-dev \
+            libmicrohttpd-dev protobuf-compiler python3-protobuf \
+            libprotobuf-dev libprotoc-dev zlib1g-dev libftdi-dev \
+            libusb-1.0-0-dev liblo-dev libavahi-client-dev python3-numpy
+      - name: Autoreconf
+        run: autoreconf -i
+      - name: Configure
+        run: ./configure --enable-rdm-tests --enable-ja-rule --enable-e133
+      - name: Build
+        run: make builtfiles VERBOSE=1
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ola-debian-stable-built-source-tree
+          path: .
+  cpplint:
+    name: cpplint
+    runs-on: ubuntu-latest
+    container: debian:stable
+    needs: build
+    steps:
+      - name: Download built source tree
+        uses: actions/download-artifact@v3
+        with:
+          name: ola-debian-stable-built-source-tree
+          path: .
+      - name: Update package database
+        run: apt-get update -y
+      - name: Install build tools
+        shell: bash
+        run: |
+          apt-get -y install pkg-config libtool autoconf \
+            automake g++ bison flex make bash-completion dh-autoreconf \
+            debhelper devscripts wget python3-pip
+      - name: Install Python lint tools
+        run: python3 -m pip install --no-input cpplint flake8
+      - name: Install build dependencies
+        shell: bash
+        run: |
+          apt-get -y install libcppunit-dev uuid-dev libncurses5-dev \
+            libmicrohttpd-dev protobuf-compiler python3-protobuf \
+            libprotobuf-dev libprotoc-dev zlib1g-dev libftdi-dev \
+            libusb-1.0-0-dev liblo-dev libavahi-client-dev python3-numpy
```

Comment:
I'm guessing this is all because we're calling configure during the make?

Given we don't (currently) do any awareness of configure about packages during cpplint, you could use the bare minimum requirements here and ./configure --disable-all-plugins etc to need fewer dependencies. Or possibly just ignore them regardless unless it re-runs configure?

Reply:
Yeah, running `make cpplint` doesn't seem to just run the cpplint command (which honestly surprised me).

What does the second `./configure` do for cpplint that isn't performed the first time? Is there a way we can disable having it call `./configure` again if it isn't necessary? Same for `make flake8`.

Reply:
I had a quick look at the Makefile and can't spot an obvious issue. Does it do the same thing for you too? I suspect a file has a different timestamp or something so it thinks it's been touched. Perhaps something around the artefact transfer?

Do you want to just make a note of this somewhere for now, maybe in the main issue as a list of things to improve in future?

Reply:
> Do you want to just make a note of this somewhere for now, maybe in the main issue as a list of things to improve in future?

Sure, I'll add it as a TODO in the overall tracking issue to circle back to once we have all the workflows minimally working.